rfc/0010-rancher-integration-policies.md

```diff
@@ -0,0 +1,162 @@
+
+- Obtain the list of available policies regardless of where they come from, and
+  their metadata to show in Rancher Explorer UI.
+- Allow for groups of policies to be installed.
+- Implement it efficiently and with low development maintenance costs.
+- Allow for airgap installations.
+
+## Examples / User Stories
+[examples]: #examples
+
+2. As a user, I want to install one Kubewarden policy owned by the
+   Kubewarden/Rancher team via Rancher Explorer UI.
+3. As a user, I want to install several Kubewarden policies owned by the
+   Kubewarden/Rancher team via Rancher Explorer UI.
+4. As a user, I want to install 3rd party Kubewarden policies via Rancher
+   Explorer UI.
+5. As a Kubewarden developer, I want to release a new Kubewarden chart/policy so
+   it can be installed via Rancher Explorer UI.
+6. As a 3rd party developer, I want to release a policy on Artifact Hub so it can
+   be installed via Rancher Explorer UI.
+
+# Detailed design
+[design]: #detailed-design
+
+## For policies owned by the Kubewarden team
+
+Follow the process listed in [RFC-9, Rancher integration of Kubewarden
+charts](./0009-rancher-integration-charts.md).
+
+Carry the Helm chart code for the policy in a `rancher-X` orphan branch, where `X`
+is the release tag of the policy being targeted.
+Using these branches allows for easy rebases and cherry-picks, and documentating
+needed changes per commit. This also separates the Rancher vendored code, which
+may be of no interest to policy authors, and may coexist with other vendors.
+
+The resulting charts from `rancher-X` will not be served and released by us, but
+used to build Rancher charts from source by using the
+[Package](https://github.com/rancher/charts/blob/dev-v2.6/docs/packages.md)
+format in https://github.com/rancher/charts.
+They will be submitted via a `packages/kubewarden/<policy>.package.yaml`, with
+`package.yaml::commit` pointing to the relevant `rancher-X` branch.
+
+For policy bundles, they can be in their own repository containing the chart
+and modifications in a `rancher-X` branch (e.g: see `kubewarden-defaults`).
+
+## For 3rd party artifacts
+
+Use the provided [artifacthub.io public API](https://artifacthub.io/docs/api/).
+Currently, the Artifact Hub API provides [`GET
+/packages/search`](https://artifacthub.io/docs/api/#/Packages/searchPackages),
+where one can set the `repository kind` to `Kubewarden policies`. With this call
+one can obtain the list of Kubewarden policies. One would then query for each of
+the policies with [GET
+/packages/kubewarden/{repoName}/{packageName}](https://artifacthub.io/docs/api/#/Packages/getKubewardenPoliciesDetails),
+and obtain its `name`, `version`, `signed` status, and `data/kubewarden-*`
+information.
+
+Artifacts owned by the Kubewarden team are also present in Artifact Hub, hence
+to avoid listing them twice, they should be skipped if `provider` of the package
+is `kubewarden`, and/or the URL starts with `https://github.com/kubewarden/`.
```

Comment:
I think this is a decision that has to be taken later on, when implementing the UI. We might want to not filter the duplicates, we should instead make clear the provenance (artifact hub vs rancher helm charts repo) because this has an impact on supportability (I'm thinking long term).

For example, Artifact Hub might have a more recent version compared to the one inside of Rancher's helm chart repository.

Reply:
Good point, amended.